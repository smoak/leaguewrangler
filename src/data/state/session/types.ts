export interface Session {
  readonly userToken: string;
}
