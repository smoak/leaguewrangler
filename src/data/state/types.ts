import { Session } from './session/types';

export * from './session/types';

export interface ApplicationState {
  readonly session: Session;
}
