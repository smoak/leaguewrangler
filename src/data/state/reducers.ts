export { default as session } from './session/reducer';
