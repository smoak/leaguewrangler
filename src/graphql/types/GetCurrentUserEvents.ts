/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

import { RsvpStatus } from "./globalTypes";

// ====================================================
// GraphQL query operation: GetCurrentUserEvents
// ====================================================

export interface GetCurrentUserEvents_currentUser_events_team {
  __typename: "Team";
  shortName: string;
  photoThumbnailUrl: string | null;
}

export interface GetCurrentUserEvents_currentUser_events_location {
  __typename: "Location";
  name: string;
  address: string | null;
  googleMapsUrl: string | null;
}

export interface GetCurrentUserEvents_currentUser_events {
  __typename: "Event";
  eventId: number;
  title: string;
  endTimestamp: number | null;
  startTimestamp: number | null;
  viewerRsvpStatus: RsvpStatus;
  team: GetCurrentUserEvents_currentUser_events_team;
  location: GetCurrentUserEvents_currentUser_events_location | null;
}

export interface GetCurrentUserEvents_currentUser {
  __typename: "User";
  events: (GetCurrentUserEvents_currentUser_events | null)[];
}

export interface GetCurrentUserEvents {
  /**
   * Gets the current user
   */
  currentUser: GetCurrentUserEvents_currentUser;
}
