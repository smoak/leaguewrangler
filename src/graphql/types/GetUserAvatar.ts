/* tslint:disable */
/* eslint-disable */
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL query operation: GetUserAvatar
// ====================================================

export interface GetUserAvatar_currentUser {
  __typename: "User";
  profilePhotoThumbnailUrl: string;
}

export interface GetUserAvatar {
  /**
   * Gets the current user
   */
  currentUser: GetUserAvatar_currentUser;
}
