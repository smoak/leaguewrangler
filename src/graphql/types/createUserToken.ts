import * as Types from './globalTypes';

export type CreateUserTokenMutationVariables = Types.Exact<{
  username: Types.Scalars['String'];
  password: Types.Scalars['String'];
}>;


export type CreateUserTokenMutation = { __typename?: 'Mutation', createUserToken?: { __typename?: 'UserToken', token: string } | null };
