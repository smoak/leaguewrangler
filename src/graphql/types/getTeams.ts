import * as Types from './globalTypes';

export type GetTeamsQueryVariables = Types.Exact<{ [key: string]: never; }>;


export type GetTeamsQuery = { __typename?: 'Query', teams: Array<{ __typename?: 'Team', name: string, id: number } | null> };
