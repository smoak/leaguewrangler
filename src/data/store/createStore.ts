import { Store, combineReducers, createStore } from 'redux';
import { composeWithDevTools } from 'redux-devtools-extension';

import * as reducers from '../state/reducers';
import { ApplicationState } from '../state/types';

const createEnhancedStore = (): Store<ApplicationState> => {
  const rootReducer = combineReducers<ApplicationState>({
    ...reducers,
  });

  const enhancer = composeWithDevTools();

  return createStore(rootReducer, enhancer);
};

export default createEnhancedStore;
