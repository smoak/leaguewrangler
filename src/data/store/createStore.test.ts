import { ApplicationState } from '../state/types';

import createStore from './createStore';

describe('createStore', () => {
  let state: ApplicationState;

  beforeEach(() => {
    const store = createStore();

    state = store.getState();
  });

  it('creates a store with no additional reducers', () => {
    expect(state).toMatchSnapshot();
  });
});
