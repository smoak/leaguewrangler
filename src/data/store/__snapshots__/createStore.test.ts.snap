// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`createStore creates a store with no additional reducers 1`] = `
Object {
  "session": Object {},
}
`;
